staging/src/k8s.io/apiserver/pkg/server/egressselector/config.go

```diff
@@ -79,97 +79,134 @@ func ValidateEgressSelectorConfiguration(config *apiserver.EgressSelectorConfigu
 	}
 	for _, service := range config.EgressSelections {
 		base := field.NewPath("service", "connection")
-		switch service.Connection.Type {
+		switch service.Connection.Protocol {
 		case "direct":
 			allErrs = append(allErrs, validateDirectConnection(service.Connection, base)...)
 		case "http-connect":
-			allErrs = append(allErrs, validateHTTPConnection(service.Connection, base)...)
+			switch service.Connection.Transport {
+			case "uds":
+				allErrs = append(allErrs, validateUDSConnection(service.Connection, base)...)
+			case "tcp":
+				allErrs = append(allErrs, validateHTTPConnection(service.Connection, base)...)
+			default:
+				allErrs = append(allErrs, field.NotSupported(
+					base.Child("protocol"),
+					service.Connection.Protocol,
+					[]string{"uds", "tcp"}))
+			}
+		case "grpc":
+			switch service.Connection.Transport {
+			case "uds":
+				allErrs = append(allErrs, validateUDSConnection(service.Connection, base)...)
+			default:
+				allErrs = append(allErrs, field.NotSupported(
+					base.Child("protocol"),
+					service.Connection.Protocol,
+					[]string{"uds"}))
+			}
 		default:
 			allErrs = append(allErrs, field.NotSupported(
-				base.Child("type"),
-				service.Connection.Type,
-				[]string{"direct", "http-connect"}))
+				base.Child("protocol"),
+				service.Connection.Protocol,
+				[]string{"direct", "http-connect", "grpc"}))
 		}
 	}
 
 	return allErrs
 }
 
 func validateDirectConnection(connection apiserver.Connection, fldPath *field.Path) field.ErrorList {
```

Comment:
We need to also check if other fields are empty. I guess you can update the validation after we nail down the API.

Reply:
Added the check. I can put in more robust tests once we nail down the API.